crates/orderbook/src/dto/order.rs

```diff
@@ -68,3 +68,34 @@ pub enum FeePolicy {
         quote: Quote,
     },
 }
+
+#[derive(Serialize, PartialEq, Debug, Clone)]
+#[cfg_attr(any(test, feature = "e2e"), derive(serde::Deserialize))]
+pub struct Solution {
```

Comment:
I think instead of just a boolean if the order was included, we should provide the executed amounts (so that in theory we can highlight how much more you got in the winning solution compared to the second best). So I'd turn the bool into an `Option<Amounts>`.

I'm not happy about exposing a public `Solution` struct with such a narrow content on the order dto module. I understand that it's inlined into Status (and thus not exposed) but in this case, can we make it private and create it from a public Status constructor?

I think we might have a proper `Solution` struct in the future which contains more stable information from the competitions table for a target tx and auciton.

Reply:
> can we make it private and create it from a public Status constructor?

Technically, we can, but what would this constructor accept? A Vec of tuples?

> So I'd turn the bool into an Option<Amounts>.

That was the initial design, which was dropped because competition json contains [Legacy orders](https://github.com/cowprotocol/services/blob/b706d23f056f0338b5896c881af735514f1279cd/crates/model/src/solver_competition.rs#L106-L126), so that requires an openapi enum. Leandro told me that FE already tracks this info which would be redundant and also would overcomplicate the response. Do we really need it?

Reply:
In the current UI proposal (https://www.figma.com/design/oSlMFMdHrELaPZACf1k7A3/Progress-Bar-Iteration?node-id=262-149&t=uTrK6IbgSRoSwQ0p-0) we are no longer showing individual amounts per solver.
This was discussed way back in the retreat/hackathon where it was raised that the winning solution does not guarantee best price for the individual order.

As for the winning one, we do need it in the last screen, but we get the order info from /orders endpoint already for other details like token addresses, limit price and so on, so it's not a must to have the amounts IMO.

Reply:
> Technically, we can, but what would this constructor accept? A Vec of tuples?

Would the `SolverCompetitionDB` type work?

> That was the initial design, which was dropped because competition json contains [Legacy orders](https://github.com/cowprotocol/services/blob/b706d23f056f0338b5896c881af735514f1279cd/crates/model/src/solver_competition.rs#L106-L126), so that requires an openapi enum

I would simply not support any legacy orders (it's ok to only support order status for new orders)

> [...] we are no longer showing individual amounts per solver.
This was discussed way back in the retreat/hackathon where it was raised that the winning solution does not guarantee best price for the individual orde

Well, we could simply not show them unless they are better

> Do we really need it?

If we are sure we will never want to display this information, sure we can leave it out. It just doesn't feel so hard to include and would allow a more delightful user experience.

Reply:
> Would the SolverCompetitionDB type work?

Not really. That means we also need to provide `order_uid` to the constructor where the business logic will leak. Maybe renaming it to SolverSolution or something similar would solve the issue with public access to the dto. Using tuples is way worse, IMO.

Reply:
> Maybe renaming it to SolverSolution

Solver solution doesn't add anything it just makes the name more vague (what other solutions are there than solver solutions). What is this thing? It's a view from the perspective of the order into the competition/solution candidates where the only thing it cares about is "am I included". Maybe something like:
- InclusionStatus
- Participation
- SolutionInclusion

all not great, but simply Solution is too broad imo.

Reply:
Renamed. Also included executed amounts into the response.